### PR TITLE
Custom SNP implementation

### PR DESCRIPTION
**Pull request by gwaygenomics**
An immediate application of `TAD_Pathways` is the ability to run a custom analysis for any input list of SNPs. The current implementation only allows for predetermined lists of SNPs (as defined by the NHGRI-EBI GWAS catalog). 

This pull request achieves a custom SNP implementation through the addition of two scripts: `scripts/build_custom_TAD_genelist.py` and `scripts/tad_util/build_snp_list.R`. I also add instructions to the README of how to implement, as well as slight modifications for dealing with updated file types and general comments/organization,

### CODE REVIEW

File: scripts/build_custom_TAD_genelist.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+output_nearest_gene_file = '{}_nearest_gene.tsv'.format(
+    os.path.splitext(output_file)[0])
+
+# Load data
+tad_genes_df = pd.read_table(os.path.join('index',
+                                          'GENE_index_hg19_hESC.tsv.bz2'),
+                             index_col=0)
+snp_df = pd.read_table(snp_data_file)
+
+results_df = pd.DataFrame(columns=tad_genes_df.columns.tolist() + ['group'])
```
Comment (tj8901nm) [inline scripts/build_custom_TAD_genelist.py]: It would be helpful if you could add more comments for lines 73-103, right now it's kind of hard to follow

File: scripts/parse_gestalt.py [hunk 2]
```diff
@@ -7,37 +7,52 @@
...
     1) Pandas Dataframe of entire TRAIT pathways
     2) Pathway summary text file
     """
 
     # Read in the entire file as string
-    with open(fh, 'r') as trait_fh:
+    with open(file_name, 'r') as trait_fh:
         gestalt = trait_fh.read()
 
+    if process:
```
Comment (tj8901nm) [inline scripts/parse_gestalt.py]: What are all these substitutions doing here?

File: scripts/tad_util/build_snp_list.R [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+#         --snp_file        <LOCATION_OF_SNP.csv>
+#         --output_file     <MAPPING_FILE.tsv>
+
+# Output:
+# A tab separated file of rsIDs mapped to genomic coordinates
+
+library("checkpoint")
+suppressMessages(checkpoint("2016-02-25"))
+
+library(biomaRt)
```
Comment (tj8901nm) [inline scripts/tad_util/build_snp_list.R]: I think it would be helpful if you use `library` import and `namespace` in a more consistent manner. In this script, I saw both `dplyr::filter` and `filter`. Also, you imported `biomaRt` and `dplyr` at the beginning, but directly used `readr` in the script.

File: scripts/tad_util/build_snp_list.R [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+                   filters = "snp_filter", values = snps[, group], mart = snp)
+  results <- results %>% dplyr::filter(!duplicated(refsnp_id))
+  results$Group <- colnames(snps)[group]
+  results_full[[group]] <- results
+}
+
+# Process results and write to file
+results_full <- do.call(rbind, results_full)
+results_full$chr_name <- as.numeric(results_full$chr_name)
+results_full <- results_full %>% filter(!is.na(results_full$chr_name))
```
Comment (tj8901nm) [inline scripts/tad_util/build_snp_list.R]: I think you don't need to specify `results_full$chr_name` here, just use `chr_name` should work fine, as you did in line52.

File: scripts/parse_gestalt.py (hunk 2 above)
Comment (gwaygenomics) [inline scripts/parse_gestalt.py]: It is a result of a very sloppy custom file that outputs as a result of a [Webgestalt analysis](http://www.webgestalt.org/).

File: scripts/tad_util/build_snp_list.R (hunk 3 above)
Comment (gwaygenomics) [inline scripts/tad_util/build_snp_list.R]: ah yes, updated in next commit!

File: scripts/tad_util/build_snp_list.R (hunk 4 above)
Comment (gwaygenomics) [inline scripts/tad_util/build_snp_list.R]: yep! nice catch.

**Comment by gwaygenomics** [comment]
ok @tj8901nm - ready for your review again. I believe I addressed all of your comments.
